scipy/cluster/_vq.pyx

```diff
@@ -232,13 +232,13 @@ def vq(np.ndarray obs, np.ndarray codes):
     outdists.fill(np.inf)
 
     if obs.dtype.type is np.float32:
-        _vq(<float32_t *>obs.data, <float32_t *>codes.data,
-            ncodes, nfeat, nobs, <int32_t *>outcodes.data,
-            <float32_t *>outdists.data)
+        _vq(<float32_t *>np.PyArray_DATA(obs), <float32_t *>np.PyArray_DATA(codes),
+            ncodes, nfeat, nobs, <int32_t *>np.PyArray_DATA(outcodes),
+            <float32_t *>np.PyArray_DATA(outdists))
```

Comment:
@scoder Cython 3.0 would make this kind of change unnecessary, so I would prefer to redo this PR once 3.0 is released.

Reply:
@mattip Should we bump the milestone again? I don't think we have Cython 3.x stable release yet? I believe there have been alpha/beta versions available via i.e., PyPI for quite some time now though.

Reply:
Yes, let's only continue on this when cpython3.0 is fully released

Reply:
As far as I can tell Cython 3.0 has not been released, so bumping the milestone per discussion above.